torchmetrics/functional/audio/sdr.py

```diff
@@ -0,0 +1,146 @@
+        keep_same_device:
+            whether to move the metric value to the device of preds
+
+    Raises:
+        ValueError:
+            If ``mir_eval`` package is not installed, or 1D input is given when compute_permutation is True
+
+    Returns:
+        sdr value of shape ``[...]`` if compute_permutation is False, else ``[..., spk]``
+        sir value of shape ``[...]`` if compute_permutation is False, else ``[..., spk]``
+        sar value of shape ``[...]`` if compute_permutation is False, else ``[..., spk]``
+
+    """
+    if not _MIR_EVAL_AVAILABLE:
+        raise ValueError(
+            "SDR metric requires that mir_eval is installed."
+            "Either install as `pip install torchmetrics[audio]` or `pip install mir_eval`"
+        )
+
+    _check_same_shape(preds, target)
+
+    if preds.ndim == 1:
+        if compute_permutation:
+            raise ValueError(
+                "SDR metric requires preds and target to be of shape [..., spk, time]"
+                " when compute_permutation is True, but 1D Tensor is given."
+            )
+
+        sdr_val_np, sir_val_np, sar_val_np, perm = bss_eval_sources(
+            target.detach().cpu().numpy()[None, ...],
+            preds.detach().cpu().numpy()[None, ...],
+            compute_permutation,
+        )
+        sdr_val = torch.tensor(sdr_val_np[0])
+        sir_val = torch.tensor(sir_val_np[0])
+        sar_val = torch.tensor(sar_val_np[0])
+    elif preds.ndim == 2:
+        preds_np = preds.detach().cpu().numpy()
+        target_np = target.detach().cpu().numpy()
+        sdr_val_np, sir_val_np, sar_val_np, perm = bss_eval_sources(target_np, preds_np, compute_permutation)
+        sdr_val = torch.tensor(sdr_val_np)
+        sir_val = torch.tensor(sir_val_np)
+        sar_val = torch.tensor(sar_val_np)
+    else:
+        preds_np = preds.reshape(-1, preds.shape[-2], preds.shape[-1]).detach().cpu().numpy()
+        target_np = target.reshape(-1, preds.shape[-2], preds.shape[-1]).detach().cpu().numpy()
+        sdr_vals, sir_vals, sar_vals = [], [], []
+        for b in range(preds_np.shape[0]):
+            output = bss_eval_sources(target_np[b], preds_np[b], compute_permutation)
+            sdr_vals.append(output[0])
+            sir_vals.append(output[1])
+            sar_vals.append(output[2])
+        sdr_val = torch.tensor(np.stack(sdr_vals))
+        sir_val = torch.tensor(np.stack(sir_vals))
+        sar_val = torch.tensor(np.stack(sar_vals))
+
+    if keep_same_device:
+        sdr_val = sdr_val.to(preds.device)
+        sir_val = sir_val.to(preds.device)
+        sar_val = sar_val.to(preds.device)
+
+    return sdr_val, sir_val, sar_val
+
+
+def sdr(preds: Tensor, target: Tensor, compute_permutation: bool = False, keep_same_device: bool = False) -> Tensor:
+    r"""sdr evaluates the Signal to Distortion Ratio (SDR) [1] metric of preds and target. sdr is a wrapper for the
+    mir_eval.separation.bss_eval_sources function.
```

Comment:
Again, please make this into a note:
https://github.com/PyTorchLightning/metrics/blob/fef83a028880f2ad3e0c265b3e5bb8a184805798/torchmetrics/audio/pesq.py#L29-L30
```suggestion
    r"""sdr evaluates the Signal to Distortion Ratio (SDR) [1] metric of preds and target.
```